lib/rucio/core/did_meta_plugins/filter_engine.py

```diff
@@ -344,13 +354,24 @@ def create_sqla_query(self, session=None, additional_model_attributes=[], additi
                         expression = or_(expression, key.is_(None))
                 elif json_column:                                                                   # -> this key filters on the content of a json column
                     if session.bind.dialect.name == 'oracle':
-                        # Compiling for Oracle yields a UnsupportedCompilationError so fall back to support for equality only.
-                        # TODO
-                        if oper != operator.eq:
-                            raise exception.FilterEngineGenericError("Oracle implementation does not support this operator.")
-                        if isinstance(value, str) and any([char in value for char in ['*', '%']]):
-                            raise exception.FilterEngineGenericError("Oracle implementation does not support wildcards.")
-                        expression = text("json_exists(meta,'$?(@.{} == \"{}\")')".format(key, value))
+                        if isinstance(value, str) and any([char in value for char in ['*', '%']]):  # wildcards
+                            if value in ('*', '%', u'*', u'%'):                                     # match wildcard exactly == no filtering on key
+                                continue
+                            else:                                                                   # partial match with wildcard == like || notlike
+                                if oper == operator.eq:
+                                    expression = text("json_exists({},'$?(@.{} like \"{}\")')".format(json_column.key, key, value.replace('*', '%')))
+                                elif oper == operator.ne:
+                                    raise exception.FilterEngineGenericError("Oracle implementation does not support this operator.")
+                        else:
+                            try:
+                                if isinstance(value, (bool)):                                       # bool must be checked first (as bool subclass of int)
+                                    expression = text("json_exists({},'$?(@.{}.boolean() {} \"{}\")')".format(json_column.key, key, ORACLE_OP_MAP[oper], value))
+                                elif isinstance(value, (int, float)):
+                                    expression = text("json_exists({},'$?(@.{} {} {})')".format(json_column.key, key, ORACLE_OP_MAP[oper], value))
+                                else:
+                                    expression = text("json_exists({},'$?(@.{} {} \"{}\")')".format(json_column.key, key, ORACLE_OP_MAP[oper], value))
```

Comment:
likewise!

Reply:
Here, same for the testing. What is maybe additional functionality here is this `.boolean()` cast which allows values to be any combination of characters that indicate boolean (e.g. `True`, `true`, `tRue`, `tRuE`, `fAlse`, `False`, `false`, etc.)